spring-cloud-starter-stream-processor-tensorflow/src/main/java/org/springframework/cloud/stream/app/tensorflow/processor/TensorflowProcessorConfiguration.java

```diff
@@ -0,0 +1,159 @@
+ *
+ * Processor uses a {@link TensorflowInputConverter} to convert the input data into TensorFlow model input format (called
+ * feeds). The input converter converts the input {@link Message} into key/value {@link Map},
+ * where the Key corresponds to a model input placeholder (feed) and the content is {@link org.tensorflow.DataType}
+ * compliant value. The default converter implementation expects either Map payload.
+ *
+ * The {@link TensorflowInputConverter} can be extended and customized.
+ *
+ * Processor's output uses the {@link TensorflowOutputConverter} to convert the computed {@link Tensor} result into a
+ * serializable message. The default implementation converts the Tensor result into {@link Tuple} triple (see:
+ * {@link TensorflowOutputConverter}).
+ *
+ * The {@link TensorflowOutputConverter} can be extended and customized to provide a convenient data representations,
+ * accustomed for a particular model
+ * (see TwitterSentimentTensorflowOutputConverter.java)
+ *
+ * @author Christian Tzolov
+ */
+@EnableBinding(Processor.class)
+@EnableConfigurationProperties(TensorflowProcessorProperties.class)
+public class TensorflowProcessorConfiguration implements AutoCloseable {
+
+	/**
+	 * Header name where the output is stored if the isSaveOutputInHeader is set
+	 */
+	public static final String TF_OUTPUT_HEADER = "TF_OUTPUT";
+
+	/**
+	 * Header name where the input is stored.
+	 * The default TensorflowInputConverter implementation will use TF_INPUT header if provided it over
+	 * the message payload.
+	 */
+	public static final String TF_INPUT_HEADER = "TF_INPUT";
+
+	private static final Log logger = LogFactory.getLog(TensorflowProcessorConfiguration.class);
+
+	@Autowired
+	private TensorflowProcessorProperties properties;
+
+	@Autowired
+	@Qualifier("tensorflowInputConverter")
+	private TensorflowInputConverter tensorflowInputConverter;
+
+	@Autowired
+	@Qualifier("tensorflowOutputConverter")
+	private TensorflowOutputConverter tensorflowOutputConverter;
+
+	@Autowired
+	private TensorFlowService tensorFlowService;
+
+	@ServiceActivator(inputChannel = Processor.INPUT, outputChannel = Processor.OUTPUT)
+	public MessageBuilder evaluate(Message<?> input) {
+
+		Map<String, Object> processorContext = new ConcurrentHashMap<>();
+
+		Map<String, Object> inputData = tensorflowInputConverter.convert(input, processorContext);
+
+		Tensor outputTensor = tensorFlowService.evaluate(
+				inputData, properties.getOutputName(), properties.getOutputIndex());
+
+		Object outputData = tensorflowOutputConverter.convert(outputTensor, processorContext);
+
+		if (properties.isSaveOutputInHeader()) {
+			// Add the inference result to the message header
+			return MessageBuilder
+					.withPayload(input.getPayload())
+					.setHeaderIfAbsent(TF_OUTPUT_HEADER, outputData);
+		}
+
+		// Add the inference result to the message payload
+		return MessageBuilder.withPayload(outputData);
```

Comment:
Sure. I've figured out how to use the `header-enricher` to achieve exactly what the `TF_OUTPUT` was meant to do. So i've removed both TF_OUTPUT and TF_INPUT (yet another not very well thought header).  I've also got rid from the MessageBuilder in favor of the plain outputData value. 
Thanks for the advise! This helped to clean redundant functionality.  
P.S. In the process i found a small bug in the `header-enricher` as you now ;)